dali/pipeline/operators/reader/loader/loader.h

```diff
@@ -107,11 +107,10 @@ class Loader {
 
   // Get a random read sample
   LoadTargetPtr ReadOne() {
-    if (lazy_init_) {
+    if (!loaded_) {
```

Comment:
I think we are checking this because checking a regular bool is cheaper than checking a std::flag_once. How much cheaper it is and whether it's worth the trouble, I don't know

Reply:
That's right
And the question here is actually: do we still need this call once if we're fully relying on these booleans?

Reply:
std::call_once is well suited for multithreading, but here all of the calls are happening on the same thread. So I'm removing it :smile:

Reply:
It seems, that `call_once` is overkill, as we don't need thread safety here, right? Maybe we can remove it?

Reply:
Done